desktop/ui/src/main/java/org/datacleaner/widgets/HdfsUrlTextFieldImpl.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.widgets;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+import org.apache.metamodel.util.HdfsResource;
+import org.datacleaner.windows.HdfsUrlChooser;
+
+public class HdfsUrlTextFieldImpl extends AbstractFilenameTextField<HdfsResource> {
```

Comment:
Naming-wise I think this is a bit busy and both adds prefix + suffix to the abstract classname. Impl isn't needed when also a prefix is there IMO. And it replaces "filename" with "url" which seems to violate inheritance (to inherit, you should be able to say that "url is a filename" which I don't think is the case).

I would normally name this: HdfsFilenameTextField.